### ISSUE TITLE
Implicitly adding the accept header `*/*` breaks some servers

### ISSUE THREAD

**Issue by tobiasbayer** [OpenFeign/feign#831]
Feign implicitly adds the accept header `*/*` to requests if none is provided.

This results in HTTP 406 on some servers if they do not provide a response content type at all (which is RFC-compliant as far as I know).

**Comment by kdavisk6** [comment]
Feign does not implicitly add any headers, the underlying `Client` implementations may do so.  Which `Client` have you configured in your builder?

**Comment by tobiasbayer** [comment]
Seems I got this wrong. Doing a quick search on the code, I found this [line](https://github.com/OpenFeign/feign/blob/9875a16f4ad3117904dde1c5a5ce3d08de7d1009/core/src/main/java/feign/Client.java#L114) and I thought it to be the culprit.
My underlying client is `OkHttp`.

**Comment by kdavisk6** [comment]
You are correct, the default `Client` does do add it.  From the comment and the history it was added because some servers were rejecting because `Accept` was missing.  Have you looked into manually configuring your OkHttp client to see if you can suppress the header?

**Comment by tobiasbayer** [comment]
From RFC 7231:

>  A request without any Accept header field implies that the user agent
   will accept any media type in response.  If the header field is
   present in a request and none of the available representations for
   the response have a media type that is listed as acceptable, the
   origin server can either honor the header field by sending a 406 (Not
   Acceptable) response or disregard the header field by treating the
   response as if it is not subject to content negotiation.

I think that by adding an accept header implicitly, the semantics of the request are changed from "I don't care about the response body" to "give me anything you have, but gibe me something".
That leads to problems with servers strictly implementing the behavior cited above when a resource does not provide a content type. Adding `*/*` as an accept header then leads to a 406.

**Comment by kdavisk6** [comment]
@tobiasbayer 

I think this leaves us in a bit of a quandary.  Are you experiencing this issue in a project you are currently working on?  Would defining your own `Client` implementation that extends from one of the existing ones work?

**Comment by tobiasbayer** [comment]
Yes, I think this would be an option. Thanks for your support.
